### PR TITLE
COMMERCE-10907 Prioritize existing portletPreferences setting over de…

### PR DESCRIPTION
**Pull request by brianikim**
…fault one

https://issues.liferay.com/browse/COMMERCE-10907

@andrea-ale-sbarra

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:commerce

**Comment by brianikim** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/shared/search/CPPriceRangeFacetsPortletSharedSearchContributor.java [hunk 1]
```diff
@@ -111,6 +116,18 @@ private Facet _getFacet(
...
+		Optional<PortletPreferences> portletPreferencesOptional =
+			portletSharedSearchSettings.getPortletPreferencesOptional();
+
+		if (portletPreferencesOptional.isPresent()) {
+			PortletPreferences portletPreferences =
+				portletPreferencesOptional.get();
+
+			rangesJSONArrayString = GetterUtil.getString(
+				portletPreferences.getValue("rangesJSONArrayString", null),
+				rangesJSONArrayString);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/shared/search/CPPriceRangeFacetsPortletSharedSearchContributor.java:128]:
```suggestion
			rangesJSONArrayString = portletPreferences.getValue("rangesJSONArrayString", rangesJSONArrayString);
```

This should do the same, can you please try it and verify?

Thank you!

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
QA passed. Will resend after writing the automation test.

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
QA passed. Will resend after writing the automation test.

**Comment by jaydawu** [comment]
Add automation and resend on https://github.com/liferay-commerce/liferay-portal/pull/3561

**Pull request closed without merging**